### DISCUSSION TITLE
Config Navmask

### DISCUSSION THREAD

**Discussion by Nghia-PD** [semuconsulting/pyubx2#171] [Q&A] [answered]
Hi everyone,

I'm trying to configurate the NAVMASK of SV, and I get ACK-NAK from the Ublox receiver:
`<UBX(ACK-NAK, clsID=CFG, msgID=CFG-VALSET)>`
Here is the function I'm trying to use
```
def generating_message(ser: serial):
    layers = pyubx2.SET_LAYER_RAM
    transaction = pyubx2.TXN_NONE
    cfgData = [("CFG_NAVMASK_SV_MASK_GAL", b"\x00")]
    msg1 = pyubx2.UBXMessage.config_set(layers=layers, transaction= transaction, cfgData=cfgData)
    ser.write(msg1.serialize())
```
Just wondering what's wrong with my code. I'm using ZED-F9T

**Comment by semuadmin** [comment]
Hi @nghiapd17012003 

What firmware version are you running on the F9T? TIM 2.20 or TIM 2.24?

I don't have an F9T to hand, but I've just tried the same on an F10T running firmware TIM 3.01 protocol 42.01 using a modified verson of the [ubxpoller](https://github.com/semuconsulting/pyubx2/blob/master/examples/ubxpoller.py) example, and I get an ACK-NAK with a mask of b"\x00", but a successful ACK-ACK with various other permissible values including b'\xff\xff\xff\xff\xff\xff\xaa\xaa' and b'\xff\xff\xff\xff\xff\xff\xff\xff' (the protocol default).

I think the problem is simply that your mask is invalid - have a look at the [Interface Manual](https://content.u-blox.com/sites/default/files/documents/u-blox-F9-TIM-2.24_InterfaceDescription_UBXDOC-963802114-13046.pdf) for your firmware version - in particular [Table 18](https://content.u-blox.com/sites/default/files/documents/u-blox-F9-TIM-2.24_InterfaceDescription_UBXDOC-963802114-13046.pdf#%5B%7B%22num%22%3A2769%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C59.527%2C495.277%2Cnull%5D) - for further details, or raise a query in the [u-blox support forum](https://portal.u-blox.com/s/).

```shell
Starting handler threads. Press Ctrl-C to terminate...
<UBX(CFG-VALSET, version=0, ram=1, bbr=0, flash=0, action=0, reserved0=0, CFG_NAVMASK_SV_MASK_GAL=b'\x00')>
<UBX(ACK-NAK, clsID=CFG, msgID=CFG-VALSET)>
Stop signal set. Waiting for threads to complete...
Processing complete
```
```shell
Starting handler threads. Press Ctrl-C to terminate...
<UBX(CFG-VALSET, version=0, ram=1, bbr=0, flash=0, action=0, reserved0=0, CFG_NAVMASK_SV_MASK_GAL=b'\xff\xff\xff\xff\xff\xff\xaa\xaa')>
<UBX(ACK-ACK, clsID=CFG, msgID=CFG-VALSET)>
Stop signal set. Waiting for threads to complete...
Processing complete
```
```shell
Starting handler threads. Press Ctrl-C to terminate...
<UBX(CFG-VALSET, version=0, ram=1, bbr=0, flash=0, action=0, reserved0=0, CFG_NAVMASK_SV_MASK_GAL=b'\xff\xff\xff\xff\xff\xff\xff\xff')>
<UBX(ACK-ACK, clsID=CFG, msgID=CFG-VALSET)>
Stop signal set. Waiting for threads to complete...
Processing complete
```

**Comment by Nghia-PD** [comment]
Hi @semuadmin,

Thank you for your answer. 
The firmware i'm using is 2.0. I have try with b"\x00\x00\x00\x00\x00\x00\x00\x00" and the default b"\xff\xff\xff\xff\xff\xff\xff\xff" and still got ACK-NAK. I have also try to use u-center to set this configuration but still not success (The other configuration still can be chanced and works). Could it possible something to do with my firmware version?

**Reply by semuadmin** [reply, marked as answer]
Question for the u-blox forum I think... I suspect it would certainly be worth upgrading to the latest firmware version (which at time of writing is 2.24).

**Reply by Nghia-PD** [reply]
I upgraded to 2.24 firmware version and got ACK-ACK,  but it seems like the GNSS is not working anymore after the update since I could not get any signal. I guess like you said, the problem come from the u-blox and should be asked in u-blox forum. 

Thanks for your help anyway!
